extensions/nodes/nodes.go

```diff
@@ -0,0 +1,92 @@
+package nodes
+
+import (
+	"errors"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/kubeapi/secrets"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	initNodeLabelKey      = "rke.cattle.io/init-node"
+	local                 = "local"
+	machinePlanSecretType = "rke.cattle.io/machine-plan"
+	machineNameSteveLabel = "rke.cattle.io/machine-name"
+)
+
+// GetInitMachine accepts a client and clusterID and returns the "init node" machine
+// object for rke2/k3s clusters
+func GetInitMachine(client *rancher.Client, clusterID string) (*v1.SteveAPIObject, error) {
+	logrus.Info("Retrieving list of secrets...")
+	secretList, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{})
```

Comment:
does listOptions have an entry for the `.Type` that you're looking for? i.e. could you just query listOptions for secret.Type=machinePlanSecretType instead of the for loop below?